### ISSUE TITLE
Inventory dataset tags are not properly munged

### ISSUE THREAD

**Issue by FuhuXia** [GSA/datagov-deploy#3481]
CKAN has a [munge_tag](https://github.com/ckan/ckan/blob/master/ckan/lib/munge.py) function that ensure tags are meeting ckan standards. Tags used in Inventory CKAN dataset are not properly munged.

For example:

![image](https://user-images.githubusercontent.com/1392461/137379191-8681e520-0e8c-4b5c-bd74-3aec95f344c6.png)

This creates multiple issues, such as searching dataset via tag by `Countries` or `countries` yields different results; when generated datajson is harvested into catalog, those improper tags will be modified by catalog CKAN, for example, `Countries => countries`, `data.gov => data-gov`

## How to reproduce

https://inventory.data.gov/dataset/data-gov-statistics-parent
or other datasets.

## Expected behavior

tags should read
`countries` `states` `statistics` `visitors` `data-gov`

## Actual behavior

`Countries` `States` `Statistics` `Visitors` `data.gov`

## Sketch
[Notes or a checklist reflecting our understanding of the selected approach]

**Comment by nickumia-reisys** [comment]
Is what `ckanext-spatial` does to allow [Uppercase characters in tags](https://github.com/GSA/datagov-deploy/issues/2413) related to this?

**Comment by FuhuXia** [comment]
> Is what `ckanext-spatial` does to allow [Uppercase characters in tags](https://github.com/GSA/datagov-deploy/issues/2413) related to this?

Nope. ckanext-spatial is not part of inventory. It should be the usmetadata extension.

**Comment by nickumia-reisys** [comment]
Hmm, well that extension has its own set of custom validation that wasn't clearly understood during the upgrade.  That should not effect current production if usmetadata was not merged into fcs.  This was the earliest [reference](https://github.com/GSA/inventory-app/pull/359/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552L17-R17) to an update that I could find.  This is the most recent update with the most [complete work](https://github.com/GSA/inventory-app/pull/370/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552L17-R17).

**Comment by FuhuXia** [comment]
Not sure. Maybe we had that issue since day 1, not related to upgrade.
